common/djangoapps/entitlements/models.py

```diff
@@ -302,6 +304,45 @@ def get_active_entitlements_for_user(cls, user):
             enrollment_course_run=None
         ).select_related('user').select_related('enrollment_course_run')
 
+    @classmethod
+    def check_for_existing_entitlement_and_enroll(cls, user, course_run_key):
+        """
+        Looks at the User's existing entitlements to see if the user already has a Course Entitlement for the course run
+        provided in the course_key.  If the user does have an Entitlement with no run set, the User is enrolled in the mode
+        set in the Entitlement.
+
+        Note: This method assumes that eligibility to enroll in the course has already been checked.
+
+        Arguments:
+            user (User): The user that we are inspecting the entitlements for.
+            course_run_key (CourseKey): The course run Key.
+
+        Returns:
+            CourseEntitlement: The CourseEntitlement that was found and used and None otherwise.
+        """
+        # Check if the User has any Entitlements that are active
+        entitlements = CourseEntitlement.get_active_entitlements_for_user(user)
```

Comment:
get_active_entitlements_for_user seems like something that should be on the user model. As in, `user.get_active_entitlements()` 
Additionally, since we are really only want fullfillable entitlements we could have a more specific 
`user.get_fullfillable_entitlements()` 
or add a optionally param to user.get_active_entitlements:
`user.get_active_entitlements(fullfillable_only=True)`